lib/src/Uncertainty/Algorithm/Simulation/NAISResult.cxx

```diff
@@ -0,0 +1,155 @@
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/NAISResult.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class NAISResult
+ */
+
+
+CLASSNAMEINIT(NAISResult);
+
+static const Factory<NAISResult> Factory_NAISResult;
+
+/* Default constructor */
+NAISResult::NAISResult()
+  : ProbabilitySimulationResult()
+{
+  //Nothing TO DO
+}
+
+/* Virtual constructor */
+NAISResult * NAISResult::clone() const
+{
+  return new NAISResult(*this);
+}
+
+// Get Input Samples
+Sample NAISResult::getAuxiliaryInputSample() const
+{
+  return auxiliaryInputSample_;
+}
+
+// Set Input Samples
+void NAISResult::setAuxiliaryInputSample(const Sample & auxiliaryInputSample)
+{
+  auxiliaryInputSample_ = auxiliaryInputSample;
+}
+
+// Get Output Samples
+Sample NAISResult::getAuxiliaryOutputSample() const
+{
+  return auxiliaryOutputSample_;
+}
+
+// Set Output Samples
+void NAISResult::setAuxiliaryOutputSample(const Sample & auxiliaryOutputSample)
+{
+  auxiliaryOutputSample_ = auxiliaryOutputSample;
+}
+
+// Get Auxiliary Density
+Distribution NAISResult::getAuxiliaryDistribution() const
+{
+  return auxiliaryDistribution_;
+}
+
+// Set Auxiliary Density
+void NAISResult::setAuxiliaryDistribution(const Distribution & auxiliaryDistribution)
+{
+  auxiliaryDistribution_ = auxiliaryDistribution;
+}
+
+// Set Maximum Outer Sampling
+void NAISResult::setMaximumOuterSampling(const UnsignedInteger & maximumOuterSampling)
+{
+  maximumOuterSampling_ = maximumOuterSampling;
+}
+
+// Get Maximum Outer Sampling
+UnsignedInteger NAISResult::getMaximumOuterSampling() const
+{
+  return maximumOuterSampling_;
+}
+
+// Set Block Size
+void NAISResult::setBlockSize(const UnsignedInteger & blocSize)
+{
+  blocSize_ = blocSize;
+}
+
+// Get Block Size
+UnsignedInteger NAISResult::getBlockSize() const
+{
+  return blocSize_;
+}
+
+// Set Variance Estimate 
+void NAISResult::setVarianceEstimate(const Scalar & varianceEstimate)
+{
+  varianceEstimate_ = varianceEstimate;
+}  
+
+// Method save() stores the object through the StorageManager
+void NAISResult::save(Advocate & adv) const
+{
+  ProbabilitySimulationResult::save(adv);
+  adv.saveAttribute("auxiliaryDistribution_", auxiliaryDistribution_);
+  adv.saveAttribute("sample_", auxiliaryInputSample_);
+}
+
+// Method load() reloads the object from the StorageManager
+void NAISResult::load(Advocate & adv)
+{
+  ProbabilitySimulationResult::load(adv);
+  adv.loadAttribute("auxiliaryDistribution_", auxiliaryDistribution_);
+  adv.loadAttribute("sample_", auxiliaryInputSample_);
+}
+
+Graph NAISResult::drawImportanceFactors() const
+{
+  throw NotYetImplementedException(HERE) << "Not implemented";
+}
+
+Scalar NAISResult::getConfidenceLength(const Scalar level) const
```

Comment:
the variable name should be commented or removed otherwise the unused variable warning turns it into an error:
```const Scalar /*level*/```